### PR TITLE
Add a Feedly account with user consent

### PR DESCRIPTION
**Pull request by kielgillard**
The Feedly integration in general is a work in progress. Adding issues for the rough edges of this PR.

### CODE REVIEW

File: NetNewsWire.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2194,16 +2200,16 @@
 				TargetAttributes = {
 					513C5CE5232571C2003D4054 = {
 						CreatedOnToolsVersion = 11.0;
-						DevelopmentTeam = SHJK2V3AJG;
+						DevelopmentTeam = 5G4WXPEDG7;
```
Comment (kielgillard) [inline NetNewsWire.xcodeproj/project.pbxproj]: Revert this change before merging.

File: Mac/Preferences/Accounts/AccountsAddViewController.swift [hunk 2]
```diff
@@ -15,7 +15,7 @@ class AccountsAddViewController: NSViewController {
 	
 	private var accountsAddWindowController: NSWindowController?
 	
-	private let addableAccountTypes: [AccountType] = [.onMyMac, .feedbin, .freshRSS]
+	private let addableAccountTypes: [AccountType] = [.onMyMac, .feedbin, .freshRSS, .feedly]
```
Comment (kielgillard) [inline Mac/Preferences/Accounts/AccountsAddViewController.swift]: Revert this change before merging, too.

File: Frameworks/Account/Account.swift [hunk 3]
```diff
@@ -307,6 +309,28 @@ public final class Account: DisplayNameProvider, UnreadCountProvider, Container,
...
 		}
 	}
+	
+	public static func oauthAuthorizationCodeGrantRequest(for type: AccountType, client: OAuthAuthorizationClient) -> URLRequest {
+		switch type {
+		case .feedly:
+			return FeedlyAccountDelegate.oauthAuthorizationCodeGrantRequest(for: client)
+		default:
+			fatalError("\(type) does not support OAuth authorization code granting.")
+		}
```
Comment (vincode-io) [inline Frameworks/Account/Account.swift]: We don't usually use fatalError and use assertionFailure instead.

Comment (kielgillard) [reply Frameworks/Account/Account.swift]: For release style builds the assertion would be stripped and I'd be required to return a value for these cases. Returning an "empty" URL request seemed like a hack so I figured it was reasonable to fatalError.

Comment (brentsimmons) [reply Frameworks/Account/Account.swift]: If I read this right: in this case it’s a serious programming error and not something that would normally happen, even in an edge case on a bad day. So I’m okay with a fatalError (or precondition check) for this.

Comment (kielgillard) [reply Frameworks/Account/Account.swift]: That's how I understand the situation, too. The programmer is trying access behaviours of certain objects that by definition they are not capable of. I expect the most basic verification of this functionality during development will have revealed this to prevent it occuring in the wild.

File: NetNewsWire.xcodeproj/project.pbxproj (hunk 1 above)
Comment (kielgillard) [reply NetNewsWire.xcodeproj/project.pbxproj]: Reverted in fc5b003ab0523160bdc56e4a5ead8c313348e845.

File: Mac/Preferences/Accounts/AccountsAddViewController.swift (hunk 2 above)
Comment (kielgillard) [reply Mac/Preferences/Accounts/AccountsAddViewController.swift]: After talking with @vincode-io, it seems there is no good reason to revert this change.
